src/screens/List/ListHiddenScreen.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react'
+import {View} from 'react-native'
+import {AppBskyGraphDefs} from '@atproto/api'
+import {msg, Trans} from '@lingui/macro'
+import {useLingui} from '@lingui/react'
+
+import {logger} from '#/logger'
+import {useGoBack} from 'lib/hooks/useGoBack'
+import {sanitizeHandle} from 'lib/strings/handles'
+import {useListBlockMutation, useListMuteMutation} from 'state/queries/list'
+import {
+  UsePreferencesQueryResponse,
+  useRemoveFeedMutation,
+} from 'state/queries/preferences'
+import {useSession} from 'state/session'
+import * as Toast from 'view/com/util/Toast'
+import {CenteredView} from 'view/com/util/Views'
+import {atoms as a, useBreakpoints, useTheme} from '#/alf'
+import {Button, ButtonText} from '#/components/Button'
+import {EyeSlash_Stroke2_Corner0_Rounded as EyeSlash} from '#/components/icons/EyeSlash'
+import {useHider} from '#/components/moderation/Hider'
+import {Text} from '#/components/Typography'
+
+export function ListHiddenScreen({
+  list,
+  preferences,
+}: {
+  list: AppBskyGraphDefs.ListView
+  preferences: UsePreferencesQueryResponse
+}) {
+  const {_} = useLingui()
+  const t = useTheme()
+  const {currentAccount} = useSession()
+  const {gtMobile} = useBreakpoints()
+  const isOwner = currentAccount?.did === list.creator.did
+  const goBack = useGoBack()
+
+  const isModList = list.purpose === 'app.bsky.graph.defs#modlist'
+
+  const [isProcessing, setIsProcessing] = React.useState(false)
+  const listBlockMutation = useListBlockMutation()
+  const listMuteMutation = useListMuteMutation()
+  const {mutateAsync: removeSavedFeed} = useRemoveFeedMutation()
+
+  const {setIsContentVisible} = useHider()
+
+  const savedFeedConfig = preferences.savedFeeds.find(f => f.value === list.uri)
+
+  const onUnsubscribe = async () => {
+    setIsProcessing(true)
+    if (list.viewer?.muted) {
+      try {
+        await listMuteMutation.mutateAsync({uri: list.uri, mute: false})
+      } catch (e) {
+        setIsProcessing(false)
+        logger.error('Failed to unmute list', {message: e})
+        Toast.show(
+          _(
+            msg`There was an issue. Please check your internet connection and try again.`,
+          ),
+        )
+        return
+      }
+    }
+    if (list.viewer?.blocked) {
+      try {
+        await listBlockMutation.mutateAsync({uri: list.uri, block: false})
+      } catch (e) {
+        setIsProcessing(false)
+        logger.error('Failed to unblock list', {message: e})
+        Toast.show(
+          _(
+            msg`There was an issue. Please check your internet connection and try again.`,
+          ),
+        )
+        return
+      }
+    }
+    Toast.show(_(msg`Unsubscribed from list`))
```

Comment:
Nit: should pry be inside the try blocks?

Reply:
Hrm, I think we just want it at the end here, since there are two separate try blocks and either of them could fail. (If both succeed we wouldn't see the toast inside of the first block)

Reply:
Better safe than sorry, sgtm